### PR TITLE
Modernizes investment price provider architecture

### LINKED ISSUE #450: Rate limiting issues with investment price retrieval
When automatic investment price retrieval is enabled, and multiple investments use Alphavantage with a free API key, then YAFFA can easily exceed the 1 request per second rate limiting, as no such control is present. Additionally, the 25 requests per day limit is also not enforced, leading to unpredictable failures.

### PR DESCRIPTION
**Pull request by kantorge**
### Overview

Introduces a robust, flexible architecture for investment price providers, replacing hardcoded logic and fields with a schema-driven, extensible system supporting per-user credentials, provider-specific settings, and generalized rate limiting. Streamlines backend and frontend integration and improves reliability, security, and maintainability.

### Key Improvements

- **Provider Flexibility**: Supports per-investment provider selection and settings via a new JSON field, enabling future providers without schema churn.
- **User-Scoped Credentials**: Adds secure, per-user provider credential storage, replacing global environment variables for API keys.
- **Dynamic Rate Limiting**: Implements generalized, provider-defined rate limiting with user plan/override support, ensuring fair, safe operation across jobs and workers.
- **Unified Retrieval Flow**: Refactors job/command dispatch to group and budget fetches by provider, handling preflight validation and error tracking.
- **Frontend Enhancements**: Adds Vue-based provider settings UI, schema-driven forms, and in-app credential management with test/fetch actions for a more intuitive user experience.
- **API and Data Model Updates**: Introduces new endpoints and models for provider configs; updates investment payloads for settings and status.
- **Migration and Backward Compatibility**: Backfills legacy scraping fields and removes old columns, ensuring a smooth transition for existing data and users.
- **Comprehensive Testing**: Expands unit and feature test coverage for new architecture, workflows, and edge cases.

### Motivation

Addresses pain points with extensibility, user credential security, and brittle provider-specific logic. Lays the foundation for supporting more providers and advanced features in the future, while reducing operational risk and complexity.

### Related

See `.ai/docs/investment-price-providers/SPECIFICATION.md` for detailed architecture and migration rationale.

## Summary by CodeRabbit

* **New Features**
  * Account "Investment provider settings" page and header link; investment form now has a "Price provider" selector with dynamic provider-specific fields, availability/status badges, a "Test Fetch" action, and server endpoints to save per‑investment provider settings; visible price‑fetch status timestamps and messages.

* **Documentation**
  * Added Investment Price Providers specification.

* **Chores**
  * CI/example env templates updated for demo provider keys; Alpha Vantage placeholder removed.

* **Tests**
  * Added extensive tests covering provider APIs, configs, console dispatching, services, models, and UI flows.

### CODE REVIEW

File: app/Console/Commands/GetInvestmentPrices.php [hunk 1]
```diff
@@ -24,20 +35,174 @@ class GetInvestmentPrices extends Command
...
+        $scanned = 0;
+        $failedPreflight = 0;
+        $dispatched = 0;
+        $skippedBudget = 0;
+
         $investments = Investment::where('auto_update', true)
             ->whereNotNull('investment_price_provider')
-            ->withMax('investmentPrices', 'date')
+            ->with(['user'])
             ->get();
```
[the remark this answers is not in the document]
Comment (kantorge) [reply app/Console/Commands/GetInvestmentPrices.php:73]: Command is scheduled once per day, and it should allow multiple manual runs.

File: app/Http/Requests/TestInvestmentPriceProviderFetchRequest.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        $resolver = app(InvestmentProviderSettingsResolver::class);
+
+        $providerSettings = $resolver->normalize(
+            $providerKey,
+            $this->input('provider_settings'),
+        );
+
+        $this->merge([
+            'provider_key' => $providerKey,
+            'symbol' => is_string($this->input('symbol')) ? mb_trim((string) $this->input('symbol')) : null,
```
[the remark this answers is not in the document]
Comment (kantorge) [reply app/Http/Requests/TestInvestmentPriceProviderFetchRequest.php:41]: PHP 8.4 is a requirement

File: app/Services/InvestmentPriceProviderContextResolver.php [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                "Investment has unknown provider: {$providerKey}",
+                $providerKey,
+                $investment->symbol
+            );
+        }
+
+        $provider = $this->providerRegistry->get($providerKey);
+        $providerMetadata = $this->providerRegistry->getMetadata($providerKey);
+
+        if (mb_trim((string) $investment->symbol) === '') {
```
[the remark this answers is not in the document]
Comment (kantorge) [reply app/Services/InvestmentPriceProviderContextResolver.php:52]: PHP 8.4 is a requirement

File: resources/views/investments/form.blade.php [hunk 4]
```diff
@@ -211,68 +211,13 @@ class="form-check-input"
...
-                        value="{{old('scrape_selector', $investment->scrape_selector ?? '' )}}"
-                    >
-                </div>
+            <div id="investmentProviderFormApp">
+                <investment-provider-form
+                    :provider-options='@json($allInvestmentPriceProviders)'
+                    :current-investment='@json($investment ?? null)'
+                    :old-input='@json(session()->getOldInput())'
+                    symbol-field-id="symbol"
+                ></investment-provider-form>
```
[the remark this answers is not in the document]
Comment (kantorge) [reply resources/views/investments/form.blade.php:220]: app/Providers/ViewServiceProvider.php loads the data
